socorro/tests/external/es/test_supersearch.py

```diff
@@ -1735,107 +1768,120 @@ def test_get_with_bad_regex(self, es_helper):
         crashstorage = self.build_crashstorage()
         api = SuperSearchWithFields(crashstorage=crashstorage)
 
-        # A bad regex kicks up a SearchParseException which supersearch converts
+        # regex is lazy-evaluated, so there has to be a crash matched by everything but
+        # the regex to produce a failure
+        now = utc_now()
+        es_helper.index_crash(
+            processed_crash={
+                "uuid": create_new_ooid(timestamp=now),
+                "signature": "test.dll",
+                "date_processed": now,
+            },
+        )
+
+        # A bad regex kicks up a shard failure which supersearch converts
         # to a BadArgumentError
         with pytest.raises(BadArgumentError):
-            api.get(signature='@"OOM | ".*" | ".*"()&%<acx><ScRiPt >sruq(9393)</ScRiPt')
-
-    def test_get_with_failing_shards(self, es_helper):
-        # NOTE(willkg): We're asserting on a url which includes the indexes being
-        # searched. If the index template includes a date, then the indexes could be in
-        # any order, so we don't include date bits and then we're guaranteed for that
-        # part of the url to be stable for mocking.
-        index_name = "testsocorro_module"
+            api.get(signature=r'@"')
+
+    def test_get_with_failing_shards(self):
+        ip, port = "127.0.0.1", 9999
         with settings.override(
             **{
-                "ES_STORAGE.options.index": index_name,
-                "ES_STORAGE.options.index_regex": f"^{index_name}$",
+                # point ES_STORAGE at fake es server location
+                "ES_STORAGE.options.url": f"http://{ip}:{port}",
             }
         ):
             crashstorage = self.build_crashstorage()
             api = SuperSearchWithFields(crashstorage=crashstorage)
 
-            with requests_mock.Mocker(real_http=False) as mock_requests:
-                # Test with one failing shard.
-                es_results = {
-                    "hits": {"hits": [], "total": 0, "max_score": None},
-                    "timed_out": False,
-                    "took": 194,
-                    "_shards": {
-                        "successful": 9,
-                        "failed": 1,
-                        "total": 10,
-                        "failures": [
-                            {
-                                "status": 500,
-                                "index": "fake_index",
+            # Test with one failing shard.
+            es_results = {
```

Comment:
> Do we know if this is what the response will look like with a failing shard? If so, how do we know that?

the test `test_get_with_bad_regex` tests a simple-to-produce shard failure where we have a special-case for raising an exception. I think I got this value by modifying that response to avoid the exception case, but I may have gotten it by manually inserting a document and then deleting data out from under es and seeing what happened.

> Also, while we're adjusting this test code, is there something we can do to make sure this test is still correct as we update Elasticsearch further in the future?
> 
> If there isn't anything we can do better here, we should have a big FIXME about how this mocked response is fragile and how we can keep it up to date manually.

I don't know a way to induce shard failures from a test case without mocking a response like this. I'll add a comment indicating that this should be manually kept up to date by modifying a bad regex response.

Reply:
oh, I know, I'll induce a bad regex response, modify it, and test with that as the mocked response